tips/TIP-0049/tip-0049.md

```diff
@@ -0,0 +1,555 @@
+---
+tip: 49
+title: IOTA 2.0 Protocol Parameters
+description: Define protocol parameters for IOTA 2.0 protocol
+author: Vivian Lin (@jkrvivian) <vivian.lin@iota.org>
+discussions-to: TODO
+status: Draft
+type: Standards
+layer: Core
+created: 2023-08-28
+---
+
+# Motivation
+
+The protocol parameters contain parameters that a node needs to apply to the components in order to communicate with
+other nodes and to process blocks properly.
+
+## Structure
+
+### Serialized layout
+
+<details>
+  <summary>Protocol Parameters</summary>
+  <blockquote>The IOTA 2.0 Protocol Parameters.</blockquote>
+</details>
+<table>
+  <tr>
+    <td>
+      <b>Name</b>
+    </td>
+    <td>
+      <b>Type</b>
+    </td>
+    <td>
+      <b>Description</b>
+    </td>
+  </tr>
+  <tr>
+    <td>Type</td>
+    <td>uint8</td>
+    <td>Set to <b>value 0</b> to denote IOTA 2.0 protocol parameters.</td>
+  </tr>
+  <tr>
+    <td>Version</td>
+    <td>uint8</td>
+    <td>The version of protocol parameters.</td>
+  </tr>
+  <tr>
+    <td>Network Name</td>
+    <td>(uint8)ByteArray</td>
+    <td>The name of the network the node is running on.</td>
+  </tr>
+  <tr>
+    <td>Bech32HRP</td>
+    <td>(uint8)ByteArray</td>
+    <td>Bech32HRP defines the HRP prefix used for Bech32 addresses in the network.</td>
+  </tr>
+  <tr>
+    <td valign="top">Storage Score Parameters <code>oneOf</code></td>
+    <td colspan="2">
+      <details open="true">
+        <summary>Storage Score Parameters</summary>
+        <blockquote>Storage Score Parameters defines the Storage Score Parameters used by a given node/network.</blockquote>
+        <table>
+          <tr>
+            <td>
+              <b>Name</b>
+            </td>
+            <td>
+              <b>Type</b>
+            </td>
+            <td>
+              <b>Description</b>
+            </td>
+          </tr>
+          <tr>
+            <td>Storage Cost</td>
```

Comment:
Can we rename this to `Storage Cost Factor` or more in line with `Factor Data` (down below) to `Factor Storage`?

Reply:
Well I do think of it as a cost, or maybe more like a price like we would say gas price in ethereum.  How about `StoragePrice`?

Reply:
I think cost is fine. It's the cost per "storage score (unit)", so I think it fits.